pageserver/src/walredo.rs

```diff
@@ -254,7 +275,218 @@ impl PostgresRedoManagerInternal {
         let start = Instant::now();
 
         let apply_result: Result<Bytes, Error>;
-        apply_result = process.apply_wal_records(tag, base_img, records).await;
+        if let ObjectTag::RelationBuffer(buf_tag) = tag {
+            apply_result = process.apply_wal_records(buf_tag, base_img, records).await;
+        } else {
+            const ZERO_PAGE: [u8; 8192] = [0u8; 8192];
```

Comment:
Well, I am also not a fan of this approach. But approach with running at page server yet another instance of non-rel-wal-redo postgres seems to me even worse.  It will be nice if we can call Postgres redo functions using FFI but I afraid that it is not possible. 

Rewriting all wal redo handler in Rust seems to be an utopia. But implementeting only non-rel WAL redo handlers may be not so awful. It is not so much lines of code and this code is not expected to be changed  frequently.

Concerning comments: sorry, will definitely add them.